labgrid/driver/mqtt.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+import queue
+import logging
+import threading
+import time
+
+import attr
+import paho.mqtt.publish as publish
+import paho.mqtt.client as mqtt
+
+from .common import Driver
+from ..factory import target_factory
+from ..protocol import PowerProtocol
+from ..step import step
+from ..util import Timeout
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class TasmotaPowerDriver(Driver, PowerProtocol):
+    bindings = {
+            "power": {"TasmotaPowerPort", "NetworkTasmotaPowerPort"}
+    }
+    delay = attr.ib(default=2.0, validator=attr.validators.instance_of(float))
+    _client = attr.ib(default=mqtt.Client(), validator=attr.validators.instance_of(mqtt.Client))
+    _status_lock = attr.ib(default=threading.Lock())
```

Comment:
Either initialize in `__attrs_post_inst__` or use `factory=`.

Reply:
`factory` is syntactic sugar for this same construct, but  I don't mind switching.

Reply:
Factory expects a function that should be called to produce the per-instance value.